lib/DDGC/Stats/Comleader.pm

```diff
@@ -0,0 +1,141 @@
+            ->search
+            ->with_created_at('>' => DateTime->now->subtract(weeks => $week))
+            ->ignore_staff_comments();
+
+        while (my $comment = $comment_rs->next) {
+            my $user = $comment->github_user->login;
+            $contributers{$user}->{both}++;
+            $contributers{$user}->{comments}++;
+            $contributers{$user}->{week}->{$week} = 1;
+        }
+
+        my $commit_rs = $self->db->resultset('GitHub::Commit')
+            ->search
+            ->with_author_date('>' => DateTime->now->subtract(weeks => $week))
+            ->ignore_staff_commits();
+
+        while (my $commit = $commit_rs->next) {
+            my $user      = $commit->github_user_author->login;
+            $contributers{$user}->{both}++;
+            $contributers{$user}->{commits}++;
+            $contributers{$user}->{week}->{$week} = 1;
+        }
+    }
+
+    # sort by commits
+    my @users = sort { ($contributers{$b}->{commits} // 0) <=> ($contributers{$a}->{commits} // 0) } keys %contributers;
+
+    for my $user (@users) {
+        my $weeks = scalar keys %{ $contributers{$user}->{week} };
+        next unless $weeks == 8;
+
+        say sprintf "username: %-30s   commits: %-5s   comments: %-5s   both: %-5s",
+            $user,
+            $contributers{$user}->{commits}  // 0,
+            $contributers{$user}->{comments} // 0,
+            $contributers{$user}->{both}     // 0;
+    }
+}
+
+sub support {
+    my ($self) = @_;
+
+    say "Support contributers with more than 50 comments in the forums";
+
+    my $rs = $self->db->resultset('Comment')
+        ->search_rs({ 
+            'me.ghosted' => 0,
+            seen_live    => 1,
+            context      => [qw/DDGC::DB::Result::Idea DDGC::DB::Result::Thread/],
+        }, { 
+            select => [{ count => 'me.id', -as => 'comment_count' }, 'users_id'],
+        })
+        ->group_by('users_id')
+        ->order_by({ -desc => 'comment_count' });
+
+    while (my $comment = $rs->next) {
+        next unless $comment->user->public;
+        next if     $comment->user->ghosted;
+        last unless $comment->get_column('comment_count') >= 50;
+
+        say sprintf "username: %-30s   comments: %-5s   email: %-50s",
+            $comment->user->username,
+            $comment->get_column('comment_count'),
+            $comment->user->email // '';
+    }
+}
+
+sub translations {
+    my ($self) = @_;
+
+    say "Translations contributers with more than 50 live translations";
+
+    my $rs = $self->db->resultset('Token::Language')
+        ->search_rs({}, {
+            select   => [ { count => 'me.id', -as => 'translation_count' }, 'translator_users_id' ],
+            group_by => 'translator_users_id',
+        })
+        ->order_by({ -desc => 'translation_count' });
+
+    while (my $token_language = $rs->next) {
+        my $user = $token_language->translator_user || next;
+
+        next unless $user->public;
+        next if $user->ghosted;
```

Comment:
Same deal here - if their translations are going live, we maybe don't care to much about the user flag.

`$user->ignore` is maybe a much stronger signal to replace this with - this happens to presistent spammers.